crates/duckdb/src/lib.rs

```diff
@@ -357,132 +257,166 @@ pub fn version() -> &'static str {
     env!("CARGO_PKG_VERSION")
 }
 
+fn to_geoarrow_record_batch(mut record_batch: RecordBatch) -> Result<RecordBatch> {
+    if let Some((index, _)) = record_batch.schema().column_with_name("geometry") {
+        let geometry_column = record_batch.remove_column(index);
+        let binary_array: GenericByteArray<GenericBinaryType<i32>> =
+            arrow::array::downcast_array(&geometry_column);
```

Comment:
Also it's cleaner to use `geometry_column.as_binary<i32>()` here: https://docs.rs/arrow/latest/arrow/array/cast/trait.AsArray.html#method.as_binary

Reply:
I used `downcast_array` because I wanted an owned array in the next line?